frontend/src/components/JaegerIntegration/JaegerScatter.tsx

```diff
@@ -89,6 +93,21 @@ class JaegerScatterComponent extends React.Component<JaegerScatterProps> {
       const isSelected = this.props.selectedTrace && trace.traceID === this.props.selectedTrace.traceID;
       const traceError = trace.spans.filter(sp => sp.tags.some(isErrorTag)).length > 0;
       const value = this.props.showSpansAverage ? averageSpanDuration(trace) || 0 : trace.duration;
+
+      var size;
+      if (this.props.provider === TEMPO) {
+        // The query to get the spans in Tempo does not return all of them,
+        // This is to avoid the resize of the circle
+        if (this.props.selectedTrace?.traceID === trace.traceID) {
+          size = this.seletedTraceMatched;
+          trace.matched = this.seletedTraceMatched;
+        } else {
+          size = trace.matched;
+        }
```

Comment:
Can you explain this in more detail?

I do not understand the setting of `size` here. I see when `provider` is not `TEMPO`, the `size` is going to be set between `MINIMAL_SIZE` and `MAXIMAL_SIZE` (which are values 2 and 30) - that is done in the line below, line 108. Here, however, I do not see any limitation on the value of `size`. How do we know this `size` does not get set to a value less than 2 or greater than 30?

Also, `size` is being set to `trace.matched` or `this.selectedTraceMatched`. I assume these are integers since they are being used to set `size`, but what do those values represent? At first I thought they were a boolean - "yes, it matched, or no it didn't match" - but that doesn't make sense since they are being set to `size`.

Reply:
The Tempo API does not return all the matched spans. But, in Tempo 2.2 it returns the `matched` field with the number of matched, like: 

```
      "spanSet": {
        "spans": [
          {
            "spanID": "2fc76fe3ac9a9ecb",
            "startTimeUnixNano": "1695630141086172000",
            "durationNanos": "782000",
            "attributes": [
              {
                "key": "service.name",
                "value": {
                  "stringValue": "details.bookinfo"
                }
              },
              {
                "key": "node_id",
                "value": {
                  "stringValue": "sidecar~10.244.1.16~details-v1-5f4d584748-2stb6.bookinfo~bookinfo.svc.cluster.local"
                }
              },
              {
                "key": "status",
                "value": {
                  "stringValue": "unset"
                }
              }
            ]
          },
          {
            "spanID": "74eeeb50a8e23208",
            "startTimeUnixNano": "1695630141083075000",
            "durationNanos": "13459000",
            "attributes": [
              {
                "key": "status",
                "value": {
                  "stringValue": "unset"
                }
              },
              {
                "key": "node_id",
                "value": {
                  "stringValue": "sidecar~10.244.2.11~productpage-v1-564d4686f-886dn.bookinfo~bookinfo.svc.cluster.local"
                }
              }
            ]
          },
          {
            "spanID": "0a09881263ee9c76",
            "startTimeUnixNano": "1695630141090410000",
            "durationNanos": "4449000",
            "attributes": [
              {
                "key": "node_id",
                "value": {
                  "stringValue": "sidecar~10.244.2.11~productpage-v1-564d4686f-886dn.bookinfo~bookinfo.svc.cluster.local"
                }
              },
              {
                "key": "status",
                "value": {
                  "stringValue": "unset"
                }
              }
            ]
          }
        ],
        "matched": 8
   }
```

So I named the same as this parameter. If it is confusing I can change it, but, not sure if it can introduce other confusions about where this value is coming from (Or we can also add a comment here to clarify) 

But you are right, I missed the MIN and MAX logic, I will add it here.